File: electronicstoreapi2/src/main/java/com/company/project/entity/Product.java
package com.company.project.entity;

import com.fasterxml.jackson.annotation.JsonBackReference;
import jakarta.persistence.*;
import lombok.Data;
import lombok.EqualsAndHashCode;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.LastModifiedDate;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

@Data
@EqualsAndHashCode(callSuper = true)
@Entity
@Table(name = "products")
@EntityListeners(AuditingEntityListener.class)
public class Product extends Auditable {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String name;

    @Column(columnDefinition = "TEXT")
    private String description;

    @Column(nullable = false, precision = 10, scale = 2)
    private BigDecimal price;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "category_id")
    @JsonBackReference
    private Category category;

    @Column(columnDefinition = "JSON")
    private String specifications;

    @ElementCollection
    @CollectionTable(name = "product_images", joinColumns = @JoinColumn(name = "product_id"))
    @Column(name = "image_url")
    private List<String> images = new ArrayList<>();

    @Enumerated(EnumType.STRING)
    private Status status = Status.ACTIVE;

    private Integer stock;

    @CreatedDate
    private LocalDateTime createdAt;

    @LastModifiedDate
    private LocalDateTime updatedAt;

    public enum Status {
        ACTIVE, INACTIVE
    }
} 

File: electronicstore2/src/types/Product.ts
// API types that match the backend models
export interface Product {
  id: number;
  name: string;
  description: string;
  price: number;
  category: Category;
  specifications: Record<string, string>;
  images: string[];
  status: 'ACTIVE' | 'INACTIVE';
  stock: number;
  createdAt: string;
  updatedAt: string;
}

export interface Category {
  id: number;
  name: string;
  description: string;
  imageUrl?: string;
  parentCategory?: Category;
  status: 'ACTIVE' | 'INACTIVE';
  createdAt: string;
  updatedAt: string;
}

// Legacy types for backwards compatibility with mock data
export interface LegacyProduct {
  id: string;
  name: string;
  category: string;
  subCategory: string;
  price: number;
  description: string;
  imageUrl: string;
  images: string[];
  specifications: Record<string, string>;
  variants?: string[];
  popularity: number;
}

export interface LegacyCategory {
  name: string;
  subCategories: string[];
  imageUrl?: string;
}

export enum ProductStatus {
  ACTIVE = 'ACTIVE',
  INACTIVE = 'INACTIVE'
} 